src/components/tabs/Tabs.tsx

```diff
@@ -1,16 +1,86 @@
-import React, { ReactNode } from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 
-export interface ITabsProps {
-  active?: boolean;
-  children: ReactNode;
-}
+import { TabsItem } from './TabsItem';
+import { ITabsProps } from './utils/tabs.interface';
+
+export const Tabs = ({ tabItems, updateSelection }: ITabsProps) => {
+  const [tabWidths, setTabWidths] = useState<
```

Comment:
what is this width-state for? is it used for the animation?

Reply:
Yes, we need to know how big each tab is in order to move the effect in the correct position. They are calculated one on rendering and they should not change anymore.